File: src/GymFlow.Application/DTOs/PhysicalAssessments/CreatePhysicalAssessmentRequest.cs
namespace GymFlow.Application.DTOs.PhysicalAssessments;

public class CreatePhysicalAssessmentRequest
{
    public DateOnly AssessmentDate { get; set; }

    public decimal WeightKg { get; set; }

    public decimal HeightCm { get; set; }

    public decimal? BodyFatPercentage { get; set; }

    public decimal? ChestCm { get; set; }

    public decimal? WaistCm { get; set; }

    public decimal? AbdomenCm { get; set; }

    public decimal? HipCm { get; set; }

    public decimal? RightArmCm { get; set; }

    public decimal? LeftArmCm { get; set; }

    public decimal? RightThighCm { get; set; }

    public decimal? LeftThighCm { get; set; }

    public decimal? RightCalfCm { get; set; }

    public decimal? LeftCalfCm { get; set; }

    public string? Notes { get; set; }
}

File: mobile/lib/features/physical_assessments/models/create_physical_assessment_request.dart
class CreatePhysicalAssessmentRequest {
  const CreatePhysicalAssessmentRequest({
    required this.assessmentDate,
    required this.weightKg,
    required this.heightCm,
    this.bodyFatPercentage,
    this.chestCm,
    this.waistCm,
    this.abdomenCm,
    this.hipCm,
    this.rightArmCm,
    this.leftArmCm,
    this.rightThighCm,
    this.leftThighCm,
    this.rightCalfCm,
    this.leftCalfCm,
    this.notes,
  });

  final DateTime assessmentDate;

  final double weightKg;
  final double heightCm;
  final double? bodyFatPercentage;

  final double? chestCm;
  final double? waistCm;
  final double? abdomenCm;
  final double? hipCm;

  final double? rightArmCm;
  final double? leftArmCm;

  final double? rightThighCm;
  final double? leftThighCm;

  final double? rightCalfCm;
  final double? leftCalfCm;

  final String? notes;

  Map<String, dynamic> toJson() {
    return {
      'assessmentDate': _formatDateOnly(assessmentDate),
      'weightKg': weightKg,
      'heightCm': heightCm,
      'bodyFatPercentage': bodyFatPercentage,
      'chestCm': chestCm,
      'waistCm': waistCm,
      'abdomenCm': abdomenCm,
      'hipCm': hipCm,
      'rightArmCm': rightArmCm,
      'leftArmCm': leftArmCm,
      'rightThighCm': rightThighCm,
      'leftThighCm': leftThighCm,
      'rightCalfCm': rightCalfCm,
      'leftCalfCm': leftCalfCm,
      'notes': notes,
    };
  }

  String _formatDateOnly(DateTime date) {
    final month = date.month.toString().padLeft(2, '0');
    final day = date.day.toString().padLeft(2, '0');

    return '${date.year}-$month-$day';
  }
}
